Libplanet.Net/HostOptions.cs

```diff
@@ -0,0 +1,53 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+
+namespace Libplanet.Net
+{
+    /// <summary>
+    /// Various options for determining <see cref="Swarm{T}"/>'s <see cref="BoundPeer"/> identity.
+    /// </summary>
+    public class HostOptions
+    {
+        public HostOptions(
+            string? host,
+            IEnumerable<IceServer> iceServers,
+            int port = 0)
+        {
+            if (port < 0)
+            {
+                throw new ArgumentException($"Given {nameof(port)} cannot be negative: {port}");
+            }
+            else if (!(host is { } ^ iceServers.Any()))
+            {
+                throw new ArgumentException(
```

Comment:
Just an idea: how about sub-classing `HostOptions` and enforcing to choose of one via type (e.g., `TurnProxyHostOptions`, `StaticHostOptions`)?

Reply:
I've had a similar idea, but wasn't sure if it'd be better to off-load type parsing responsibility to the client. As it stands, thought it'd be easier from a user standpoint to just dump all arguments and have the library to tell early on that certain combinations of arguments can't be used together. I think this can be later further refactored to include the said functionality/implementation when SDK/client API gets more streamlined. 🙄

Reply:
I guess that we can offer a parsing feature even with these types if we need it. (e.g., `HostOptions.Create()`). 

and... I also think that it isn't parsing actually because options for the relay server and its host don't share the same representation.

Reply:
Eh, tried implementing `TrunProxyOptions` and `StaticHostOptions` using `HostOptions.Create()`, but seemed like a bit of an overkill. 😐

I have some vague idea about creating an additional layer for handling `HostOptions` where type handling of subclasses of the `HostOptions` class might be more appropriate. Imma kick this can down the road a bit more. 😗